pkg/controllermanager/controller/seed/lifecycle/reconciler.go

```diff
@@ -125,15 +98,15 @@ func (c *livecycleReconciler) Reconcile(ctx context.Context, req reconcile.Reque
 	bldr.WithMessage("Gardenlet stopped posting seed status.")
 	if newCondition, update := bldr.WithNowFunc(metav1.Now).Build(); update {
```

Comment:
I think it would be better to pass a func that uses the `r.Click.Now` function here.

TBH I was thinking of opening an issue/PR to deprecate the `WithNowFunc` for the condition builder and replace it with a `WithClock` func. `WithNowFunc` even has a wrong comment specifying that it is intended to be used for tests, but we actually use it in multiple places.

Reply:
I agree and I even started doing this as well, but then I reverted these changes because I thought it's probably better/cleaner to put this into a dedicated PR. Feel free to open an issue/PR for it, this would definitely be a great improvement!